packages/twenty-server/src/engine/workspace-manager/dev-seeder/data/services/timeline-activity-seeder.service.ts

```diff
@@ -0,0 +1,395 @@
+      await this.getObjectMetadataIds(workspaceId);
+
+    let activityIndex = 0;
+
+    const entityConfigs = [
+      { type: 'company', seeds: COMPANY_DATA_SEEDS },
+      { type: 'person', seeds: PERSON_DATA_SEEDS },
+      { type: 'note', seeds: NOTE_DATA_SEEDS },
+      { type: 'task', seeds: TASK_DATA_SEEDS },
+      { type: 'opportunity', seeds: OPPORTUNITY_DATA_SEEDS },
+    ];
+
+    entityConfigs.forEach(({ type, seeds }) => {
+      seeds.forEach((seed, index) => {
+        const activity = this.createTimelineActivity(type, seed, index);
+
+        timelineActivities.push(activity);
+        activityIndex++;
+
+        // For notes and tasks, create additional linked timeline activities on target objects
+        if (type === 'note' || type === 'task') {
+          const linkedObjectMetadataId =
+            type === 'note' ? noteMetadataId : taskMetadataId;
+          const linkedActivities = this.createLinkedTimelineActivities(
+            type,
+            seed,
+            index,
+            activityIndex,
+            linkedObjectMetadataId,
+          );
+
+          timelineActivities.push(...linkedActivities);
+          activityIndex += linkedActivities.length;
+        }
+      });
+    });
+
+    if (timelineActivities.length > 0) {
+      const batchSize = 100;
+      const timelineActivityBatches = chunk(timelineActivities, batchSize);
+
+      for (const batch of timelineActivityBatches) {
+        await entityManager
+          .createQueryBuilder(undefined, undefined, undefined, {
+            shouldBypassPermissionChecks: true,
+          })
+          .insert()
+          .into(`${schemaName}.timelineActivity`, [
+            'id',
+            'name',
+            'properties',
+            'linkedRecordCachedName',
+            'linkedRecordId',
+            'linkedObjectMetadataId',
+            'workspaceMemberId',
+            'companyId',
+            'personId',
+            'noteId',
+            'taskId',
+            'opportunityId',
+            'createdAt',
+            'updatedAt',
+            'happensAt',
+          ])
+          .orIgnore()
+          .values(batch)
+          .execute();
+      }
+    }
+  }
+
+  private createTimelineActivity(
+    entityType: string,
+    entitySeed: Record<string, unknown>,
+    index: number,
+  ): Record<string, unknown> {
+    const generateTimelineActivityId = (type: string, idx: number): string => {
+      const prefix = '20202020';
+      const entityCodes: Record<string, string> = {
+        company: '0001',
+        person: '0201',
+        note: '0601',
+        task: '0651',
+        opportunity: '0851',
+      };
+      const code = entityCodes[type] || '0000';
+      const paddedIndex = String(idx).padStart(4, '0');
+
+      return `${prefix}-${code}-4000-8001-${paddedIndex}00000001`;
+    };
+
+    const creationDate = new Date().toISOString();
+
+    const timelineActivity: Record<string, unknown> = {
```

Comment:
Record<string, unknown> is not a great type here we know more about timelineActivities